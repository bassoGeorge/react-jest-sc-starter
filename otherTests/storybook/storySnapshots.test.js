import initStoryshots from "@storybook/addon-storyshots";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import React from "react";

/* React test renderer has an issue with portals. Seems like react testing library also screws up a bit */
beforeAll(() => {
  ReactDOM.createPortal = jest.fn((element, node) => {
    return element;
  });
});

afterEach(() => {
  ReactDOM.createPortal.mockClear();
});

initStoryshots({
  // In case you need to wait a tick, turn asyncJest true and in the test function, call done() when its done
  asyncJest: false,
  test: ({ story, context, done }) => {
    const StoryElement = story.render;
    const { container } = render(<StoryElement />);
    expect(container.firstChild).toMatchSnapshot();
  }
});
