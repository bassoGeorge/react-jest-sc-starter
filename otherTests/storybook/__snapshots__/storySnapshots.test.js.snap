// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots My Button Basic Usage 1`] = `
.c0 {
  border-radius: 8px;
  border: none;
  background: red;
  color: white;
  padding: 1.5em;
  font-size: 18px;
}

<button
  class="c0"
>
  Trying it out
</button>
`;

exports[`Storyshots My Button Clicks 1`] = `
.c0 {
  border-radius: 8px;
  border: none;
  background: red;
  color: white;
  padding: 1.5em;
  font-size: 18px;
}

<button
  class="c0"
>
  Click Me
</button>
`;
